interview-questions.md

```diff
@@ -7,53 +7,55 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a hash in Ruby?
 
-  Your answer:
+  Your answer: A hash in ruby is a collection of data, which can be a mix between any of the primitive data types like strings, numbers, booleans, ect.. To have a hash you need to have a key:value pairs. Which basically means you need a unique name for the key so you can call upon on it, and a value which can be any primitive data type. Values, unlike keys, don't have to unique to be used.
 
-  Researched answer:
+  Researched answer: A unique key:value pairs. Instead of calling upon an index, you would call upon a the unique key. Now if you attempted to call upon a key that doesn't exist, all you would get in return is 'nil'.
 
 
 
 2. What is a gem?
 
-  Your answer:
+  Your answer: A gem is a file library that has preloaded files like rspec. Gem's are usefull when it comes to coding because the developer dosen't have to waste time making possibly hundreds of files of code just to get a project set up. They are a lifesaver as a developer when it comes to meeting deadlines of projects 
 
-  Researched answer:
+  Researched answer: Gem's aare commonly shared and distrubuted among other ruby users to extend the capability of their application, or to use as reusable code inside of their own rails projects.
 
 
 
 3. What is Ruby on Rails?
 
-  Your answer:
+  Your answer: Ruby on Rails is an open source web development tool that uses the Ruby program to develop applications. It's similar to React.js in the way that the code you write can be updated to the webpage it is assigned to. Both also have a local server function so you can see what you're doing live to the webpage.
 
-  Researched answer:
+  Researched answer: Ruby on Rails is a full stack framework that focus on developer usablility. Ontop of rending a full webpage and the code that runs it, Ruby on Rails also has a basic security functionality built in it.
 
 
 
 4. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: I'm not sure what a relational database is, I'll have to research that question afterwards. I do know that databases in general are there to store large amounts of information. For instance you can make a database that stores all the user information of people who sign up to your website or other user based product.
```

Comment:
Awesome!!  I really like how you shifted what you do not know off the top of your head to what you do know, well done!